src/pages/AMM/Pools/components/PoolsInsights/PoolsInsights.tsx

```diff
@@ -47,40 +96,73 @@ const PoolsInsights = ({ pools, accountPoolsData, refetch }: PoolsInsightsProps)
 
-  const transaction = useTransaction(Transaction.AMM_CLAIM_REWARDS, {
-    onSuccess: refetch
-  });
-
-  const handleClickClaimRewards = () => accountPoolsData && transaction.execute(accountPoolsData.claimableRewards);
+  const handleClickClaimRewards = () => setOpen(true);
 
   const hasClaimableRewards = totalClaimableRewardUSD > 0;
+
+  const isBtnDisabled = isTrasanctionFormDisabled(form, transaction.fee);
+
   return (
-    <Dl wrap direction='row'>
-      <Card flex='1'>
-        <DlGroup direction='column' alignItems='flex-start' gap='spacing1'>
-          <StyledDt color='primary'>{t('supply_balance')}</StyledDt>
-          <StyledDd color='secondary'>{supplyBalanceLabel}</StyledDd>
-        </DlGroup>
-      </Card>
-      <Card flex='1'>
-        <DlGroup direction='column' alignItems='flex-start' gap='spacing1'>
-          <StyledDt color='primary'>{t('total_liquidity')}</StyledDt>
-          <StyledDd color='secondary'>{totalLiquidityUSD}</StyledDd>
-        </DlGroup>
-      </Card>
-      <Card direction='row' flex='1' gap='spacing2' alignItems='center' justifyContent='space-between'>
-        <DlGroup direction='column' alignItems='flex-start' gap='spacing1'>
-          <StyledDt color='primary'>{t('rewards')}</StyledDt>
-          <StyledDd color='secondary'>{formatUSD(totalClaimableRewardUSD, { compact: true })}</StyledDd>
-        </DlGroup>
-        {hasClaimableRewards && (
-          <AuthCTA onPress={handleClickClaimRewards} loading={transaction.isLoading}>
-            Claim
-          </AuthCTA>
-        )}
-      </Card>
-    </Dl>
+    <>
+      <Dl wrap direction='row'>
+        <Card flex='1'>
+          <DlGroup direction='column' alignItems='flex-start' gap='spacing1'>
+            <StyledDt color='primary'>{t('supply_balance')}</StyledDt>
+            <StyledDd color='secondary'>{supplyBalanceLabel}</StyledDd>
+          </DlGroup>
+        </Card>
+        <Card flex='1'>
+          <DlGroup direction='column' alignItems='flex-start' gap='spacing1'>
+            <StyledDt color='primary'>{t('total_liquidity')}</StyledDt>
+            <StyledDd color='secondary'>{totalLiquidityUSD}</StyledDd>
+          </DlGroup>
+        </Card>
+        <Card direction='row' flex='1' gap='spacing2' alignItems='center' justifyContent='space-between'>
+          <DlGroup direction='column' alignItems='flex-start' gap='spacing1'>
+            <StyledDt color='primary'>{t('rewards')}</StyledDt>
+            <StyledDd color='secondary'>{totalClaimableRewardUSDLabel}</StyledDd>
+          </DlGroup>
+          {hasClaimableRewards && (
+            <CTA onPress={handleClickClaimRewards} loading={transaction.isLoading}>
+              Claim
+            </CTA>
+          )}
+        </Card>
+      </Dl>
+      <Modal
+        isOpen={isOpen}
+        onClose={() => setOpen(false)}
+        shouldCloseOnInteractOutside={(el) => !overlappingModalRef.current?.contains(el)}
```

Comment:
It's not very clear why this condition applies, could you put it into separate function with explanatory name that would be passed here.

Reply:
I will create a hook for this. For now I will leave a comment.

Reply:
I think we can achieve it without hook, just a function that takes ref and element as parameters would suffice.